processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -1670,6 +1755,20 @@ public Type getReturnTypeToConstruct() {
         return returnTypeToConstruct;
     }
 
+    public boolean hasSubClassMappings() {
+        return !subClasses.isEmpty();
+    }
+
+    public List<SubClassMapping> constructSubClassMappings(List<Parameter> parameters) {
+        subClasses.forEach( subClass -> subClass.updateWithParameters( parameters ) );
+        return subClasses;
+    }
```

Comment:
We strive for the models to be immutable if possible. This code is invoked from the `BeanMappingMethod.ftl`. Can we do this in a different place and only get the `subClassMappings` here?